File: WabiLyticsFunction/src/main/java/wabilytics/CarrierChangedFunction.java
package wabilytics;

import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.io.IOException;
import java.net.URL;
import java.util.HashMap;
import java.util.Map;
import java.util.stream.Collectors;

import com.amazonaws.services.lambda.runtime.Context;
import com.amazonaws.services.lambda.runtime.RequestHandler;
import com.amazonaws.services.lambda.runtime.LambdaLogger;
import com.j256.ormlite.dao.Dao;
import com.j256.ormlite.dao.DaoManager;
import com.j256.ormlite.jdbc.JdbcConnectionSource;
import com.j256.ormlite.support.ConnectionSource;

/**
 * Handler for requests to Lambda function.
 */
public class CarrierChangedFunction implements RequestHandler<Object, Object> {

    private static Dao<Eventos, String> accountDao;

    private static void initialize(Context context) {
        final LambdaLogger logger = context.getLogger();
        if (accountDao != null) return;
        logger.log("initializing function...\r\n");
        final String databaseUrl = context.getClientContext().getEnvironment().get("db.url");
        final String username = context.getClientContext().getEnvironment().get("db.username");
        final String password = context.getClientContext().getEnvironment().get("db.password");
        //final String databaseUrl = "jdbc:mysql://192.168.0.66:3306/prueba";
        try {
            ConnectionSource connectionSource = new JdbcConnectionSource(databaseUrl, username, password);
            accountDao = DaoManager.createDao(connectionSource, Eventos.class);
            logger.log("function initialization done.\r\n");
        } catch (Exception e) {
            logger.log(String.format("Error of Type %s: %s", e.getClass().getName(), e.getMessage()));
            e.printStackTrace();
        }
    }

    public Object handleRequest(final Object input, final Context context) {
        initialize(context);
        final LambdaLogger logger = context.getLogger();
        logger.log("CarrierChangedFunction invoked\r\n");
        Map<String, String> headers = new HashMap<>();
        headers.put("Content-Type", "application/json");
        headers.put("X-Custom-Header", "application/json");
        try {
            final String pageContents = this.getPageContents("https://checkip.amazonaws.com");
            String output = String.format("{ \"message\": \"hello world\", \"location\": \"%s\" }", pageContents);
            final Eventos eventos = new Eventos(output);
            accountDao.create(eventos);
            return new GatewayResponse(output, headers, 200);
        } catch (Exception e) {
            logger.log(String.format("Error of Type %s: %s", e.getClass().getName(), e.getMessage()));
            return new GatewayResponse("{}", headers, 500);
        }
    }

    private String getPageContents(String address) throws IOException{
        URL url = new URL(address);
        try(BufferedReader br = new BufferedReader(new InputStreamReader(url.openStream()))) {
            return br.lines().collect(Collectors.joining(System.lineSeparator()));
        }
    }
}

File: src/main/kotlin/wabilytics/carrierChanged/CarrierChangedFunction.kt
package wabilytics.carrierChanged

import com.amazonaws.services.lambda.runtime.Context
import com.amazonaws.services.lambda.runtime.RequestHandler
import wabilytics.EventRequest
import java.util.*

/**
 * Handler for requests to Lambda function.
 */
class CarrierChangedFunction : RequestHandler<Any?, String> {

    override fun handleRequest(input: Any?, context: Context): String {
        CarrierDao.initialize(context)
        val logger = context.logger
        logger.log("CarrierChangedFunction invoked - version 1.0\r\n")
        logger.log(String.format("input type: %s\r\n",input?.javaClass?.name))
        logger.log(String.format("input: %s\r\n",input?.toString()))
        val eventRequest = EventRequest.fromMap(input as Map<String, Object>?)
        return try {
            val carrier = Carrier(UUID.randomUUID().toString(), eventRequest.payload.toString(), true, false)
            CarrierDao.save(carrier)
            "OK"
        } catch (e: Exception) {
            logger.log(String.format("Error of Type %s: %s", e.javaClass.name, e.message))
            //rethrow in order to dispatch message to dead letter queue
            throw e
        }
    }
}